### ISSUE TITLE
How to i install this?

### ISSUE THREAD

**Issue by Oxi84** [Dao-AILab/flash-attention#1050]
Hello, how can I install this without pip? From source?

**Comment by janEbert** [comment]
Yes, see at the bottom of this section in the main README: https://github.com/Dao-AILab/flash-attention/blob/7ef24848cf2f855077cef88fe122775b727dcd74/README.md#flashattention-3-beta-release
```shell
cd hopper
python setup.py install
```

**Comment by Oxi84** [comment]
Do you mean?

git clone https://github.com/Dao-AILab/flash-attention/
cd hopper
python setup.py install

**Comment by janEbert** [comment]
Yes, exactly! To be really pedantic, you'd have to do `cd flash-attention/hopper` instead of `cd hopper` if following your commands exactly. :)
